sqlalchemy_utils/functions/database.py

```diff
@@ -441,59 +459,46 @@ def database_exists(url):
 
     """
 
-    def get_scalar_result(engine, sql):
-        result_proxy = engine.execute(sql)
-        result = result_proxy.scalar()
-        result_proxy.close()
-        engine.dispose()
-        return result
-
-    def sqlite_file_exists(database):
-        if not os.path.isfile(database) or os.path.getsize(database) < 100:
-            return False
-
-        with open(database, 'rb') as f:
-            header = f.read(100)
-
-        return header[:16] == b'SQLite format 3\x00'
-
     url = copy(make_url(url))
     database, url.database = url.database, None
-    engine = sa.create_engine(url)
+    dialect_name = url.get_dialect().name
 
-    if engine.dialect.name == 'postgresql':
+    if dialect_name == 'postgresql':
         text = "SELECT 1 FROM pg_database WHERE datname='%s'" % database
-        return bool(get_scalar_result(engine, text))
-
-    elif engine.dialect.name == 'mysql':
+        if databases is None:
+            databases = ('postgres', 'template0', 'template1', None)
```

Comment:
If `url` passed into this function provided a db name, for example:
```
postgresql://user:pass@1.2.3.4/dbname
```

Wouldn't we want to check `dbname` first here?

Reply:
Ok nevermind, I see that the SQL command provided by `text` will handle that.